controllers/confidential_handler.go

```diff
@@ -0,0 +1,132 @@
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *KataConfigOpenShiftReconciler) updateConfigMap(cmName string, namespace string, newData map[string]string) error {
+	// Get current configMap.
+	configMap := &corev1.ConfigMap{}
+	if err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      cmName,
+		Namespace: namespace,
+	}, configMap); err != nil {
+		return err
+	}
+
+	update := false
+
+	// Loop over each new value
+	// Update the value if it's different.
+	// Log the change.
+	for key, newValue := range newData {
+		if configMap.Data[key] != newValue {
+			r.Log.Info("Updating configMap", "name", cmName, "key", key, "value", newValue)
+			configMap.Data[key] = newValue
+			update = true
+		}
+	}
+
+	if update {
+		// Update the configMap on Kubernetes.
+		return r.Client.Update(context.TODO(), configMap)
+	} else {
+		return nil
+	}
+}
+
+// enableFeatureConfidential sets config maps to confidential values.
+//
+// Changes the ImageConfigMap, so that the image creation job will create a confidential image.
+// This will happen at the first reconciliation loop, before the image creation job starts.
+//
+// Changes the peer pods configMap to enable confidential.
+// This will happen likely after several reconciliation loops, because it has prerequisites:
+//
+//   - Peer pods must be enabled in the KataConfig.
+//   - The peer pods config map must exist.
+func (r *KataConfigOpenShiftReconciler) enableFeatureConfidential() error {
+
+	// ImageConfigMap
+
+	if err := InitializeImageGenerator(r.Client); err != nil {
+		return err
+	}
+	ig := GetImageGenerator()
+
+	if ig.provider == unsupportedCloudProvider {
+		r.Log.Info("unsupported cloud provider, skipping confidential image configuration")
+	} else {
+		if ig.isImageIDSet() {
+			r.Log.Info("Image ID is already set, skipping confidential image configuration")
+		} else {
+			// Create ImageConfigMap, if it doesn't exist already.
+			if err := ig.createImageConfigMapFromFile(); err != nil {
+				return err
+			}
+
+			// Patch ImageConfigMap.
+			imageConfigMapData := map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "yes"}
+			if err := r.updateConfigMap(ig.getImageConfigMapName(), OperatorNamespace, imageConfigMapData); err != nil {
+				return err
+			}
+		}
+	}
+
+	// peer pods config
+
+	if r.kataConfig.Spec.EnablePeerPods {
+		// Patch peer pods configMap, if it exists.
+		peerpodsCMData := map[string]string{"DISABLECVM": "false"}
+		if err := r.updateConfigMap(peerpodsCMName, OperatorNamespace, peerpodsCMData); err != nil {
+			if k8serrors.IsNotFound(err) {
+				// ConfigMap doesn't exist yet, will try again at the next reconcile run.
+			} else {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// disableFeatureConfidential resets config maps to non-confidential values.
+func (r *KataConfigOpenShiftReconciler) disableFeatureConfidential() error {
```

Comment:
would it be more logically sense to enable/disable the image CM from within ig.updateImageConfigMap based on if confidential is enabled/disabled?
And similarly for the peer-pods-cm at the processKataConfigInstallRequest maybe
Sorry for the late suggestion, feel free to ignore it.. :)

Reply:
Yes, but that would mean integrating the feature in the bulk of the code and not keeping it in a separate file. I understand that the whole point of this feature gates thing is to keep things separate, or isn't it?

Reply:
Right, well, the question is if editing the same objects in multiple locations/phases of the reconcile loop for the sake of code separation wroth it, IDK..
We are at the point we can still track it, so i guess it's fine, however, we do expect more features that touches the same CMs (automatic setting of peer-pods-cm, GPU enablement etc..), we'll have to make it more generic some point..